cranelift/isle/isle/src/parser.rs

```diff
@@ -12,12 +12,21 @@ pub fn parse(lexer: Lexer) -> Result<Defs> {
     parser.parse_defs()
 }
 
+/// Parse without positional information. Provided mainly to support testing, to
+/// enable equality testing on structure alone.
+pub fn parse_without_pos(lexer: Lexer) -> Result<Defs> {
+    let mut parser = Parser::new(lexer);
+    parser.disable_pos();
+    parser.parse_defs()
+}
+
```

Comment:
I really don't like this, but I also couldn't think of a better way. Please let me know if you have an alternative suggestion.

* Having `parse_without_pos` as a public function is a bit distasteful. Previously I had `parse_without_pos` in the test binary instead, but that requires you to export `Parser` itself, and that seems worse.
* We can't guard `parse_without_pos` with `#[cfg(test)]` because then it's not visible to an integration test binary.
* Potentially the printer tests could be structured as unit tests instead of modeled off the `run_tests.rs` binary, which would allow use of conditional compilation for test only.
* We could have an `eq_without_pos` on AST types instead. However, I assumed that would require a lot of hand- or macro-generated boilerplate.

If `parse_without_pos` really is the best option, I also wondered if it would be nicer to instead augment the `parse` method with a `Positions` enum argument to configure whether positions are populated or dropped.

Reply:
A final option: a helper that maps through all AST nodes and sets positions to e.g. zero? That seems easiest to keep up to date and least likely to be subtly incorrect (vs. hand-written equality predicates), especially given compile errors on new AST node kinds.